neuro_evolution_ctrnn/brains/i_pytorch_brain.py

```diff
@@ -0,0 +1,111 @@
+                attr_weight_ih_li = "weight_ih_l{}".format(i)
+                attr_weight_hh_li = "weight_hh_l{}".format(i)
+
+                weight_ih_li = getattr(self.brain, attr_weight_ih_li)
+                weight_hh_li = getattr(self.brain, attr_weight_hh_li)
+
+                weight_ih_li_size = np.prod(weight_ih_li.size())
+                weight_hh_li_size = np.prod(weight_hh_li.size())
+
+                # Do not forget to reshape back again
+                weight_ih_li.data = torch.from_numpy(
+                    individual[current_index: current_index + weight_ih_li_size]).view(weight_ih_li.size()).float()
+                current_index += weight_ih_li_size
+
+                weight_hh_li.data = torch.from_numpy(
+                    individual[current_index: current_index + weight_hh_li_size]).view(weight_hh_li.size()).float()
+                current_index += weight_hh_li_size
+
+                if config.use_bias:
+                    attr_bias_ih_li = "bias_ih_l{}".format(i)
+                    attr_bias_hh_li = "bias_hh_l{}".format(i)
+
+                    bias_ih_li = getattr(self.brain, attr_bias_ih_li)
+                    bias_hh_li = getattr(self.brain, attr_bias_hh_li)
+
+                    bias_ih_li_size = bias_ih_li.size()[0]
+                    bias_hh_li_size = bias_hh_li.size()[0]
+
+                    bias_ih_li.data = torch.from_numpy(
+                        individual[current_index: current_index + bias_ih_li_size]).float()
+                    current_index += bias_ih_li_size
+
+                    bias_hh_li.data = torch.from_numpy(
+                        individual[current_index: current_index + bias_hh_li_size]).float()
+                    current_index += bias_hh_li_size
+            self.weight_ho = np.array(
+                individual[current_index: current_index + self.output_size * config.hidden_size]).reshape(
+                (self.output_size, config.hidden_size))
+            current_index += self.output_size * config.hidden_size
+            assert current_index == len(individual)
+
+            self.hidden = self.get_hidden(config)
+
+    @classmethod
+    def get_individual_size(cls, config: ConfigClass, input_space: Space, output_space: Space):
+        num_layers = config.num_layers
+        number_gates = cls.get_number_gates()
+        hidden_size = config.hidden_size
+        index = 0
+        # size of the learnable input-hidden weights
+
+        index += hidden_size * IBrain._size_from_space(input_space) * number_gates
+        if num_layers > 1:
+            index += hidden_size * hidden_size * (num_layers - 1) * number_gates
+
+        # size of the learnable hidden-hidden weights
+        index += hidden_size * hidden_size * num_layers * number_gates
+
+        if config.use_bias:
+            index += 2 * hidden_size * num_layers * number_gates
+        index += IBrain._size_from_space(output_space) * hidden_size
+        return index
```

Comment:
This needs to be fixed before merge. At least for the LSTM this does not calculate the correct number of parameters. For example: It does not loop through the `num_layers`. A correct implementation is in the LSTMNumPy class

Reply:
The both versions have different individual sizes, because Pytorch uses twice as much biases.
Also pytorch can only handle "width and height" of the neurons. The numpy version can handle individual layer sizes. 
The pytorch version doesn't run through all layers, because each layer (except the first one) are the same in shape and indivdual size.